src/parquet/arrow/reader.h

```diff
@@ -99,10 +99,18 @@ class PARQUET_EXPORT FileReader {
   // Read column as a whole into an Array.
   ::arrow::Status ReadColumn(int i, std::shared_ptr<::arrow::Array>* out);
 
-  // Read a table of flat columns into a Table.
+  // Reads a specific top level schema field into an Array
+  ::arrow::Status ReadSchemaField(int i, std::shared_ptr<::arrow::Array>* out);
+
+  // Reads a specific top level schema field into an Array
+  // Read only the indicated leaf columns
```

Comment:
Yes, I wasn't sure about this too - I think that there should be a decision about terminology which also derives the APIs.
cc @advancedxy